### DISCUSSION TITLE
Feature list questions

### DISCUSSION THREAD

**Discussion by phirestalker** [nextcloud/recognize#274] [General]
How up to date is the feature list in the Readme? I believe I remember that there was work on the feature to recognize faces that were not in the contacts. Am I right, and if so, how far along is it?

Also, I was wondering if there is any hope of recognizing pets, as in individual pets. I'm sure it is not possible by their faces, but perhaps by their markings?

**Comment by marcelklehr** [comment]
Hello @phirestalker 

Recognizing faces that are not in the contacts is now possible in the latest version.

Recognizing pets is an interesting feature. I can investigate a little to see if there are models that allow this type of classification.

**Reply by phirestalker** [reply]
I know Google Photos can recognize pets, as I have mine named and automatically added to a shared album of pets. I am unsure of how they do it, and I doubt they are willing to share.

**Reply by marcelklehr** [reply]
It might be possible to use [this animal classification model](https://tfhub.dev/google/experts/bit/r50x1/in21k/animal/1) to extract features of the animal from photos that include animals and then cluster animal photos by those features.

**Comment by phirestalker** [comment]
I want to use this instead of Photoprism, etc. It does everything important that other photo-management projects do. The only thing missing is handling duplicates. I realize that this is simply tagging photos in Nextcloud, but could there be a way to tag photos with some hash that can later be compared to find duplicates?

The hash would need to be based on hamming distance or another comparable hash. I am not sure what mechanism could be used to later search for duplicates, so I am just putting it out there.

**Reply by marcelklehr** [reply]
There is an app that can find duplicates for you: https://apps.nextcloud.com/apps/duplicatefinder

**Reply by phirestalker** [reply]
That app relies on exact bit-for-bit duplicates. I was hoping for something specifically for images based on perceptual duplicates. Thanks for the info though.

**Reply by IrArevalo00** [reply]
You might want to check out [mediaDC](https://nextcloud.com/blog/introducing-mediadc-the-app-that-helps-you-find-duplicated-media-files/) if you haven't already

**Comment by IrArevalo00** [comment]
Hello,

I would like to know if face recognition is feasible with local external storage?
From what I've seen, users can give names to faces but just with their own photos. 
Recognize does tag external storage photos, but how would someone go about giving names to faces on external storage shares?

**Reply by marcelklehr** [reply]
External storage should be supported out of the box. If that doesn't work, please open an issue.

**Reply by stweil** [reply]
That's a problem in our installation. Several users have configured the same huge external storage with lots of images, and all those images are processed for every user separately. Is it possible disable the support for external storage? Currently we had to disable the recognize app completely. :-(

**Reply by marcelklehr** [reply]
> and all those images are processed for every user separately

They are not. Recognize scans each storage only once.

**Comment by KoooDT** [comment]
How can I use another language in Tags?
All tags are in English, but I want to use another language

**Reply by marcelklehr** [reply]
Starting with Nextcloud 25.0.1 tags should be displayed in the language the user selected in the settings.

**Reply by KoooDT** [reply]
Thanks a lot :)
